### DISCUSSION TITLE
Open source 1D simulations of divertor detachment

### DISCUSSION THREAD

**Discussion by jonmaddock** [ukaea/PROCESS#3487] [Ideas]
In GitLab by @mkovari on Dec 18, 2020, 12:14

[The role of particle, energy and momentum losses in 1D simulations of divertor detachment](https://iopscience.iop.org/article/10.1088/1361-6587/ab1321/meta)  
BD Dudson et al.

>  A new 1D divertor plasma code, SD1D, has been used to examine the role of recombination, radiation, and momentum exchange in detachment. 

>  A system of determining the detached condition is developed such that the divertor solution at a given Ttarget (or lack of one) is determined by the simultaneous solution of two equations for target ion current — one dependent on power losses and the other on momentum. There are regions of Ttarget where there is no solution and the plasma 'jumps' from high to low Ttarget states. The novel analysis methods developed here provide an intuitive way to understand complex detachment phenomena...

They have found two equations for the target flux as a function of temperature based on power balance and momentum balance. The intersections of the two curves readily predict available steady-state solutions. 

This probably contains the same physics as the Kallenbach model, but has perhaps been studied more thoroughly.

**The SD1D code is available at https://github.com/boutproject/SD1D. All inputs, data and processing scripts are available at DOI [10.5281/zenodo.1410281](https://doi.org/10.5281/zenodo.1410281).**

Probably we should try to incorporate ths in PROCESS.

@ajpearcey @jmorris-uk @skahn ?

**Comment by jonmaddock** [comment]
In GitLab by @skahn on Dec 21, 2020, 18:23

This is worth having a read indeed !     
The main question is probably how the divertor geometry is included as ST divertor may be super-X :)

**Comment by jonmaddock** [comment]
In GitLab by @mkovari on Apr 25, 2022, 09:08

I note that Ben Dudson and others are still improving the open source 1D divertor code, SD1D - see, for example, [Investigation of the role of hydrogen molecules in 1D simulation of divertor detachment](https://scholar.google.co.uk/scholar_url?url=https://iopscience.iop.org/article/10.1088/1361-6587/ac6827/pdf&hl=en&sa=X&d=2749029506284998867&ei=Ma5jYoiUE8LZmQHnraWYCA&scisig=AAGBfm2mC59BYIO1xvgx_KEmuPNHEs2SCg&oi=scholaralrt&hist=mAF8-9sAAAAJ:13469018716134826043:AAGBfm3H-wBxkfCZ6Y20qU6esHnpU8DQNw&html=&pos=0&folt=kw).

Any comments, @ajpearcey, @jmorris-uk, @skahn, @ym1906  ?

**Comment by mkovari** [comment]
There is another 1D SOL model: [Multi-machine benchmark of the self-consistent 1D scrape-off layer model DIV1D from stagnation point to target with SOLPS-ITER](https://iopscience.iop.org/article/10.1088/1361-6587/ad2e37/meta).

Stefan Mijin and David Moulton are co-authors.

It is not open source, but the paper says that "The DIV1D source code is available upon reasonable request from the authors."

**Comment by jmorris-uk** [comment]
SD1D is not longer maintained and the README recommends using Hermes 3 instead (also managed by Ben Dudson). Both codes are GPL 3.0, not sure how that interfaces with our MIT and LGPL 2.1 licences. 

https://github.com/bendudson/hermes-3/?tab=readme-ov-file

This is an identical discussion to #3046. We shouldn't implement multiple different 1-D divertor models right now, a more in-depth review is needed before proceeding with which one to use.

**Comment by j-a-foster** [comment]
Another issue related to this was #1201.

**Comment by mkovari** [comment]
[A simple, accurate model for detachment access](https://scholar.google.co.uk/scholar_url?url=https://arxiv.org/pdf/2504.05486&hl=en&sa=X&d=5214776812117029976&ei=WMv4Z9noC9SyieoP2ane-QQ&scisig=AFWwaeZ_ZlWe4M_qXNLDwOJiKSiC&oi=scholaralrt&hist=mAF8-9sAAAAJ:11336155548172897273:AFWwaeYawouV9RUh72VPH8XO0HMQ&html=&pos=4&folt=kw)
T Body, A Kallenbach, T Eich - arXiv preprint arXiv:2504.05486, 2025

The software developed for this paper will be available at [github.com/cfs-energy/extended_lengyel](github.com/cfs-energy/extended_lengyel).

@jonmaddock @chris-ashe @jmorris-uk @je-cook @clmould @timothy-nunn

**Comment by geograham** [comment]
New location: https://github.com/FusionDLS/FusionDLS
